interpreter/interpreter.py

```diff
@@ -31,6 +31,36 @@ def interpret_internal(ast, assignment_store):
                             pass
             case 'assign-statement':
                 assignment_store[ast[1]] = interpret_internal(ast[2], assignment_store)
+                
+            case 'assign-function':
+                parameter_list = interpret_internal(ast[2], None)
```

Comment:
There should not be a None here. If you call the function with a variable as input, the recursive function call will try to get its value in the assignment store.

Reply:
but this is just for expanding the parameter list in the declaration, which should not have any variables, just names for the parameters